### PR TITLE
LLM Plugin config and device

### PR DESCRIPTION
**Pull request by rasapala**
### 🛠 Summary

JIRA CVS-143019
Added handling of device and plugin configuration to llm calculator options.
--device - "CPU"
--plugin_config - {"MODEL_DISTRIBUTION_POLICY":"TENSOR_PARALLEL","INFERENCE_NUM_THREADS":12}

### 🧪 Checklist

- [x] Unit tests added.
- [x] The documentation updated.
- [x] Change follows security best practices.
``

### CODE REVIEW

File: src/llm/llmnoderesources.hpp [hunk 1]
```diff
@@ -47,6 +47,8 @@ struct LLMNodeResources {
 public:
     std::shared_ptr<ContinuousBatchingPipeline> cbPipe = nullptr;
     std::string modelsPath;
+    std::string device;
+    std::string pluginConfig;
```
Comment (mzegla) [inline src/llm/llmnoderesources.hpp:51]: This type is not in line with CB lib. Shouldn't we have ov::AnyMap here and do some parsing of the string that comes from node options?

File: docs/llm_calculator.md [hunk 2]
```diff
@@ -59,12 +59,14 @@ node: {
...
+-    optional uint64 cache_size = 3 [default = 4];
 -    optional uint64 block_size = 4 [default = 32];
 -    optional uint64 max_num_seqs = 5 [default = 256];
--    optional bool dynamic_split_fuse = 7 [default = false];
-
+-    optional bool dynamic_split_fuse = 7 [default = true];
+-    optional string device = 8 [default = "CPU"]
+-    optional string plugin_config = 9 [default = ""]
 
+It is important to set cache_size to bigger values for bigger models because the default value of 4GB is just used for simple testing.
```
Comment (dtrawins) [inline docs/llm_calculator.md:69]:
```suggestion
The value of `cache_size` might have performance  implications. It is used for storing LLM model KV cache data. Adjust it based on the model size and the expected level of concurrency. Set `50` or even more for high load with large models.
```

File: third_party/llm_engine/llm_engine.bzl [hunk 3]
```diff
@@ -19,8 +19,8 @@ def llm_engine():
     llm_engine_repository(name="_llm_engine")
     new_git_repository(
         name = "llm_engine",
-        remote = "https://github.com/mzegla/openvino.genai",
-        commit = "582af04ae297a5ebc8246e326979495ab2e12748",
+        remote = "https://github.com/rasapala/openvino.genai",
```
Comment (atobiszei) [inline third_party/llm_engine/llm_engine.bzl:22]: Is it supposed to be different fork or is it just for test?

**Review by atobiszei — APPROVED** [review]

File: third_party/llm_engine/llm_engine.bzl (hunk 3 above)
Comment (rasapala) [reply third_party/llm_engine/llm_engine.bzl:22]: Need to merge a PR with review from Ilya to switch back to his repository.

File: src/llm/llmnoderesources.hpp (hunk 1 above)
Comment (rasapala) [reply src/llm/llmnoderesources.hpp]: Fixed.

File: third_party/llm_engine/llm_engine.bzl (hunk 3 above)
Comment (mzegla) [reply third_party/llm_engine/llm_engine.bzl:22]: PR merged, let's switch to the main fork:
https://github.com/ilya-lavrenov/openvino.genai/commits/ct-beam-search/

File: docs/llm_calculator.md [hunk 4]
```diff
@@ -59,12 +59,14 @@ node: {
 The calculator supports the following node_options for tuning the pipeline configuration:
 -    required string models_path = 1;
 -    optional uint64 max_num_batched_tokens = 2 [default = 256];
--    optional uint64 num_kv_blocks = 3 [default = 364];
+-    optional uint64 cache_size = 3 [default = 4];
```
Comment (mzegla) [inline docs/llm_calculator.md:62]:
```suggestion
-    optional uint64 cache_size = 3 [default = 8];
```

**Review by mzegla — APPROVED** [review]

**Review by dtrawins — APPROVED** [review]

**Pull request merged**